tutorials/batch-approve-swap.md

```diff
@@ -0,0 +1,310 @@
+Now we can deploy the `SimpleDex` contract using the `run` command and specifying `moonbase` as the network:
+
+```
+npx hardhat run --network moonbase scripts/deploy.js
+```
+
+!!! note
+    If you want to run the script in a standalone fashion using `node <script>`, you'll need to require the Hardhat Runtime Environment explicitly using `const hre = require("hardhat");` in the `deploy.js` file.
+
+After a few seconds, the contract will be deployed, and you should see the address in the terminal. We'll need to use the address in the following sections to interact with the contract, so make sure you save it.
+
+## Swap Tokens {: #swapping-tokens }
+
+With the contract deployed, now we can create a script that will enable us to get started by swapping DEV tokens for DTOK tokens. Once we have the DTOKs, we can get into the approval and swap. We'll take a quick look at how the approval and swap work normally before diving into using the batch precompile to batch these transactions. 
+
+For simplicity, we'll create a single script to handle all of the logic needed to swap DEV to DTOKs and back, called `swap.js`. We'll add this file to the `scripts` directory:
+
+```
+touch swap.js
+```
+
+### Create Contract Instances {: #create-contract-instances }
+
+We'll need to create contract instances for each of our contracts so that we can access each contract's functions. For this, we're going to use the `getContractAt()` helper function of the Hardhat plugin.
+
+For this step, we're going to need the contract address of the `SimpleDex` contract. Then we'll be able to use the `SimpleDex` contract instance to retrieve the `DemoToken` contract address through the `token()` function.
+
+We'll also need to add a contract instance for the batch precompile, which is located at `{{ networks.moonbase.precompiles.batch }}`.
+
+You can add the following code to the `swap.js` file:
+
+```js
+const simpleDexAddress = "INSERT-ADDRESS-OF-DEX";
+
+async function main() {
+  // Create instance of SimpleDex.sol
+  const simpleDex = await ethers.getContractAt(
+    "SimpleDex",
+    simpleDexAddress
+  );
+
+  // Create instance of DemoToken.sol
+  const demoTokenAddress = await simpleDex.token();
+  const demoToken = await ethers.getContractAt(
+    "DemoToken",
+    demoTokenAddress
+  );
+
+  // Create instance of Batch.sol
+  const batchAddress = "{{ networks.moonbase.precompiles.batch }}";
+  const batch = await ethers.getContractAt("Batch", batchAddress);
+}
+main();
+```
+
+### Add Check Balances Helper Function {: #add-function-to-check-balances }
+
+Next, we're going to create a helper function that will be used to check the balance of DTOK tokens the DEX and the signer account has. This will be particularly useful to see balance changes after the swaps are complete.
+
+In this function, you'll call the `balanceOf()` function of the `DemoToken` contract, passing in the address of the signer and the DEX, and then print the formatted results in DTOKs to the terminal:
```

Comment:
I would state that the contract has an ERC-20 interface so you can check the balance using the normal `balanceOf` function